### DISCUSSION TITLE
Steting up C/C++ in vscodium (without vscode)

### DISCUSSION THREAD

**Discussion by renben-dev** [VSCodium/vscodium#1665] [Q&A]
Hi fellas,

well I've been searching the web for these answers for a week, so so little out there, Hope you can help me on the following:
(I've VSCodium on linux, kernel 6.1. The only extension installed is CodeLLDB)
1. in C++ how do I set up a task.json so Ctrl+Alt+B builds?
2. How do I setup a launch.json?

What extensions do I need to make it work (I only know how to with cppbuild, which is propretary so there's no point. I really want to avoid going back to VScode only because I don't know I can't find these basic setups which apparently are so difficult to find a tutorila for),

Thanks,

**Comment by mofosyne** [comment]
Was trying to follow https://www.youtube.com/watch?v=8jLOx1hD3_o but using codium and hit this same problem of trying to figure out how to run an example code that has this as it's tasks.json

```
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "cppbuild",
			"label": "Build with Clang",
			"command": "/usr/bin/clang++-12",
			"args": [
				"-g",
				"-std=c++20",
				"${workspaceFolder}/*.cpp",
				"-o",
				"${fileDirname}/rooster"
			],
			"options": {
				"cwd": "${fileDirname}"
			},
			"problemMatcher": [
				"$gcc"
			],
			"group": "build",
			"detail": "compiler: /usr/bin/clang++-12"
		}
	]
}
```

Looks like cppbuild is from possibly microsoft intellisense? Anyway requesting advice on how to deal with such (Or how to port it to something that is cross compatible with vscode and vscodium. (E.g. An extension with support for cppbuild settings?

Looking forward for advice! In the meantime switching to vscode to be able to learn these tutorials.

**Reply by DanielRein-1** [reply]
Bro am on linux using vscodium when i run files that have math.h i get an error

**Comment by natelwhite** [comment]
Came across this thread when looking into intellisense options for C++ in vscodium. I had the same issue, but was alerted to change "cppbuild" to "process". Hopefully this helps.

**Reply by DanielRein-1** [reply]
am on linux using vscodium when i run files that have math.h i get an error how can i solve this to link my code with the math library

**Reply by natelwhite** [reply]
That sounds like a compiling error, so that's not really related to vscodium. I suggest looking over your build process and which header you should be using.

**Reply by DanielRein-1** [reply]
I figured it out anyway it's not a compiling error

**Comment by renben-dev** [comment]
> Came across this thread when looking into intellisense options for C++ in vscodium. I had the same issue, but was alerted to change "cppbuild" to "process". Hopefully this helps.

Oh my god! This is good and unexpected after this long time, it's crazy I couldn't get around it. I'll try it out as soon as I go back home. Let you know
